Source: Ali Polatel <alip@exherbo.org>
Upstream: https://bugzilla.samba.org/show_bug.cgi?id=7514
          http://article.gmane.org/gmane.network.rsync.general/21814
Reason: Fix socketpair_tcp()
--- rsync-3.0.7/socket.c	2009-01-17 21:46:11.000000000 +0000
+++ rsync-3.0.7/socket.c	2010-06-15 19:44:39.000000000 +0000
@@ -761,6 +761,7 @@
 	sock2.sin_len = sizeof sock2;
 #endif
 	sock2.sin_family = PF_INET;
+	sock2.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
 
 	bind(listener, (struct sockaddr *)&sock2, sizeof sock2);
 
@@ -783,7 +784,7 @@
 	} else
 		connect_done = 1;
 
-	if ((fd[0] = accept(listener, (struct sockaddr *)&sock, &socklen)) == -1)
+	if ((fd[0] = accept(listener, (struct sockaddr *)&sock2, &socklen)) == -1)
 		goto failed;
 
 	close(listener);
